code/modules/mob/living/simple_animal/friendly/farm_animals.dm

```diff
@@ -48,7 +47,7 @@
 
 /mob/living/simple_animal/hostile/retaliate/goat/Life()
 	. = ..()
-	if(.)
+	if(. && sentience_type != SENTIENCE_BOSS)

```

Comment:
This makes me wonder why exactly the king goat is a goat subtype. What behaviour exactly do you get from goats that you wouldn't get from retaliate? Is the vine eating and milking part actually critical to what you are trying to do? This just seems to make the code worse for little to no reason.
If anything the vine eating and milking stuff should be turned into components since milking is actually something that's shared between quite a few mobs.